prime-router/docs/design/proposals/translation-template-test-plan.md

```diff
@@ -0,0 +1,57 @@
+# Translation Template Test Plan
+
+## Problem Statement
+As a Reportstream engineer, I want tests that verify that the changes I make to translations will not break essential 
+rules set forth by the spec or our users. 
+
+Currently, we use integration tests for this purpose, but these do not give us the ability to easily tell which thing 
+broke/changed or even really what is specifically being tested.
+
+## Considerations
+- We not only want to check the rules set forth in the spec, but the specific ones created for senders and receivers.
+
+## Proposal on *what* to test
+We should write tests when:
+- The spec specifies a condition.
+- There is a requirement specified by a sender or receiver. This means that there needs to be a test for every rule in
+  a sender or receiver transform, This includes the default transform since these are ReportStream specific rules we
+  are implementing
+  - This can be done by comparing segments rather than field by field.
+- We have created an extension not specified in the spec.
+- We also want to test the losslessness of an HL7 message being transformed into a FHIR message and then back to an
+HL7 message. Losslessness beign defined as the data being in the same place it was when originally passed in.
+
+## Options for *how* to test 
```

Comment:
Overall, I think the how might end up being an implementation detail.  `integration` vs `unit` might not be a great differentiator at least how the code is currently configured.